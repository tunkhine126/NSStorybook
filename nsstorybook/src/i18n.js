/**
 * i18n.js
 *
 * This will setup the i18n language files and locale data for your app.
 *
 *   IMPORTANT: This file is used by the internal build
 *   script `extract-intl`, and must use CommonJS module syntax
 *   You CANNOT use import/export in this file.
 */
const addLocaleData = require('react-intl').addLocaleData; //eslint-disable-line
const enLocaleData = require('react-intl/locale-data/en');
const esLocaleData = require('react-intl/locale-data/es');
const ptLocaleData = require('react-intl/locale-data/pt');

// Custom Locale-Data for Haitian / Creole (not supported by react-intl)
const htLocaleData = {
  locale: 'ht',
  pluralRuleFunction: (e, t) => (t && e === 1 ? 'one' : 'other'),
};

const combineTranslationMessages = require('./translations/langs.json');

addLocaleData(enLocaleData);
addLocaleData(esLocaleData);
addLocaleData(ptLocaleData);
addLocaleData(htLocaleData);

const DEFAULT_LOCALE = 'en';

// prettier-ignore
const appLocales = [
  'en',
  'es',
  'pt',
  'ht'
];

const formatTranslationMessages = (locale, messages) => {
  const defaultFormattedMessages =
    locale !== DEFAULT_LOCALE
      ? formatTranslationMessages(DEFAULT_LOCALE, combineTranslationMessages)
      : {};

  const flattenFormattedMessages = (formattedMessages, key) => {
    const formattedMessage =
      !messages[key] && locale !== DEFAULT_LOCALE
        ? defaultFormattedMessages[key]
        : messages[key];
    return Object.assign(formattedMessages, {
      [key]: formattedMessage[locale],
    });
  };
  return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

const translationMessages = {
  en: formatTranslationMessages('en', combineTranslationMessages),
  es: formatTranslationMessages('es', combineTranslationMessages),
  pt: formatTranslationMessages('pt', combineTranslationMessages),
  ht: formatTranslationMessages('ht', combineTranslationMessages),
};

exports.appLocales = appLocales;
exports.formatTranslationMessages = formatTranslationMessages;
exports.translationMessages = translationMessages;
exports.DEFAULT_LOCALE = DEFAULT_LOCALE;
