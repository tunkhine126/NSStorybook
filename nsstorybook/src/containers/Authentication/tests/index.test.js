/**
 *
 * Tests for Authentication
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import { ApolloProvider } from 'react-apollo';

import client from 'utils/apollo-client';

import { DEFAULT_LOCALE } from '../../../i18n';
import configureStore from 'configureStore';
import { Authentication } from '../index';

describe('<Authentication />', () => {
  let store;
  const history = {
    location: {
      pathname: '/dashboard',
    },
  };

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ApolloProvider client={client}>
            <Authentication dispatch={dispatch} history={history} />
          </ApolloProvider>
        </IntlProvider>
      </Provider>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const dispatch = jest.fn();
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ApolloProvider client={client}>
            <Authentication dispatch={dispatch} history={history} />
          </ApolloProvider>
        </IntlProvider>
      </Provider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
