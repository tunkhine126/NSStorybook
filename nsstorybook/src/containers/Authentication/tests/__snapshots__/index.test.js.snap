// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authentication /> Should render and match the snapshot 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-auto"
    >
      <img
        alt="New Story Logo"
        class="makeStyles-bottomMargin-308"
        src="IMAGE_MOCK"
        width="75"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        <span>
          Log In
        </span>
      </h5>
    </div>
    <form>
      <div
        class="MuiGrid-root makeStyles-form-317 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="email"
            id="email-label"
          >
            <span>
              Email*
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="email"
              name="email"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="password"
            id="password-label"
          >
            <span>
              Password*
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="password"
              name="password"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
      >
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-537 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-540"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <span>
              Remember Me
            </span>
          </span>
        </label>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-306 makeStyles-bottomMargin-308 makeStyles-topMargin-309 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            <span>
              Log In
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-307 MuiTypography-colorPrimary"
          href="/"
        >
          <span>
            Forgot your password?
          </span>
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-307 MuiTypography-colorPrimary"
          href="/"
        >
          <span>
            Didn't receive unlock instructions?
          </span>
        </a>
      </div>
    </form>
  </div>
</div>
`;
